### PR TITLE
Added Campaign & My pledges Screen in User Portal

### LINKED ISSUE #2070: USER: Add Campaign & My Pledges Screen in user portal
**Is your feature request related to a problem? Please describe.**
The Admin Portal supports almost all aspects of Funds/Campaigns/Pledges but User portal doesn't have any screens to access Funds/Campaigns/Pledges.

The User portal misses following screen/features:
- **Campaigns screen** displaying all campaigns ongoing in organization.
- Functionality to **make a pledge** for a campaign.
- **My Pledges** screen showing all pledges and mock functionality to pay certain amount.
- All the search/sort functionality required for above screens.

**Describe the solution you'd like**
- Add Campaigns Tab in SideNav which will redirect to Campaigns Screen displaying all ongoing campaigns in organization.
- Add functionality to pledge certain amount towards a campaign via the Campaign Screen
- Add My Pledges Tab in SideNav which will redirect to My Pledges Screen displaying all the pledges 
- Add mock functionality to pay certain amount of a pledge

### PR DESCRIPTION
**Pull request by GlenDsza**
**What kind of change does this PR introduce?**

bugfix, feature, refactoring

**Issue Number:**

Fixes #2070 

**Did you add tests for your changes?**

Yes 

**Snapshots/Videos:**

https://github.com/user-attachments/assets/1cf54a79-3a9e-4643-9a20-58a581320acc

**Summary**

- Added Campaign Screen in User Portal, to display all campaigns belonging to specific organization
- Added My Pledges Screen, where user can view his/her pledges, while having option to create/edit/delete them.
- Above screens features option for sorting, searching and search by filters.
- Updated CreateFundRaisingCampaign Mutation to support [API PR](https://github.com/PalisadoesFoundation/talawa-api/pull/2441)
- Added relevant translations and test cases.
- Restructured Parent-child  component in User Portal

**Does this PR introduce a breaking change?**

Yes,
This PR should be merged only after [API PR](https://github.com/PalisadoesFoundation/talawa-api/pull/2441) is merged

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Enhanced localization support with new error messages and translated entries for various user interface elements across multiple languages, including English, French, Hindi, Spanish, and Chinese.
	- Introduced new components for managing campaigns and pledges, improving user engagement and functionality.
	- Added routes for a more intuitive navigation experience within the application.

- **Bug Fixes**
	- Improved error handling for loading failures with new error messages in localization files.

- **Style**
	- Updated and added new CSS styles to enhance the visual consistency of components related to campaigns and pledges.

- **Tests**
	- Implemented comprehensive unit tests for new components and functionalities, ensuring reliable performance across various scenarios.

### CODE REVIEW

**Comment by GlenDsza** [comment]
The introspection currently fails as queries and mutation are update as per [API PR](https://github.com/PalisadoesFoundation/talawa-api/pull/2441).
This PR should only be merged when related API PR is once merged

File: src/state/reducers/userRoutersReducer.test.ts [hunk 1]
```diff
@@ -15,6 +15,8 @@ describe('Testing Routes reducer', () => {
         { name: 'People', url: 'user/people/undefined' },
         { name: 'Events', url: 'user/events/undefined' },
         { name: 'Donate', url: 'user/donate/undefined' },
+        { name: 'Campaigns', url: 'user/campaigns/undefined' },
+        { name: 'My Pledges', url: 'user/pledges/undefined' },
```
[the remark this answers is not in the document]
Comment (GlenDsza) [reply src/state/reducers/userRoutersReducer.test.ts:19]: The test case is correct

**Comment by Cioppolo14** [comment]
@GlenDsza Is it expected for the introspection test to fail?

**Comment by GlenDsza** [comment]
> @GlenDsza Is it expected for the introspection test to fail?

@Cioppolo14 Yes for the moment its expected to fail as the changes in this PR are with respect to changes made in [API PR](https://github.com/PalisadoesFoundation/talawa-api/pull/2441) which is yet to be merged. Once API PR is merged, the introspection test will pass on re-running test workflows

**Comment by tasneemkoushar** [comment]
@GlenDsza approving the pr, make sure to merge api PR before merging this one

**Review by tasneemkoushar — APPROVED** [review]

**Pull request merged**